op-batcher/batcher/service.go

```diff
@@ -0,0 +1,351 @@
+		SeqWindowSize:      bs.RollupConfig.SeqWindowSize,
+		ChannelTimeout:     bs.RollupConfig.ChannelTimeout,
+		MaxChannelDuration: cfg.MaxChannelDuration,
+		SubSafetyMargin:    cfg.SubSafetyMargin,
+		MaxFrameSize:       cfg.MaxL1TxSize - 1, // subtract 1 byte for version
+		CompressorConfig:   cfg.CompressorConfig.Config(),
+	}
+	if err := bs.Channel.Check(); err != nil {
+		return fmt.Errorf("invalid channel configuration: %w", err)
+	}
+	return nil
+}
+
+func (bs *BatcherService) initTxManager(cfg *CLIConfig) error {
+	txManager, err := txmgr.NewSimpleTxManager("batcher", bs.Log, bs.Metrics, cfg.TxMgrConfig)
+	if err != nil {
+		return err
+	}
+	bs.TxManager = txManager
+	return nil
+}
+
+func (bs *BatcherService) initPProf(cfg *CLIConfig) error {
+	if !cfg.PprofConfig.Enabled {
+		return nil
+	}
+	log.Debug("starting pprof server", "addr", net.JoinHostPort(cfg.PprofConfig.ListenAddr, strconv.Itoa(cfg.PprofConfig.ListenPort)))
+	srv, err := oppprof.StartServer(cfg.PprofConfig.ListenAddr, cfg.PprofConfig.ListenPort)
+	if err != nil {
+		return err
+	}
+	bs.pprofSrv = srv
+	log.Info("started pprof server", "addr", srv.Addr())
+	return nil
+}
+
+func (bs *BatcherService) initMetricsServer(cfg *CLIConfig) error {
+	if !cfg.MetricsConfig.Enabled {
+		bs.Log.Info("metrics disabled")
+		return nil
+	}
+	m, ok := bs.Metrics.(opmetrics.RegistryMetricer)
+	if !ok {
+		return fmt.Errorf("metrics were enabled, but metricer %T does not expose registry for metrics-server", bs.Metrics)
+	}
+	bs.Log.Debug("starting metrics server", "addr", cfg.MetricsConfig.ListenAddr, "port", cfg.MetricsConfig.ListenPort)
+	metricsSrv, err := opmetrics.StartServer(m.Registry(), cfg.MetricsConfig.ListenAddr, cfg.MetricsConfig.ListenPort)
+	if err != nil {
+		return fmt.Errorf("failed to start metrics server: %w", err)
+	}
+	bs.Log.Info("started metrics server", "addr", metricsSrv.Addr())
+	bs.metricsSrv = metricsSrv
+	return nil
+}
+
+func (bs *BatcherService) initDriver() {
+	bs.driver = NewBatchSubmitter(DriverSetup{
+		Log:          bs.Log,
+		Metr:         bs.Metrics,
+		RollupCfg:    bs.RollupConfig,
+		Cfg:          bs.BatcherConfig,
+		Txmgr:        bs.TxManager,
+		L1Client:     bs.L1Client,
+		L2Client:     bs.L2Client,
+		RollupClient: bs.RollupNode,
+		Channel:      bs.Channel,
+	})
+}
+
+func (bs *BatcherService) initRPCServer(cfg *CLIConfig) error {
+	server := oprpc.NewServer(
+		cfg.RPC.ListenAddr,
+		cfg.RPC.ListenPort,
+		bs.Version,
+		oprpc.WithLogger(bs.Log),
+	)
+	if cfg.RPC.EnableAdmin {
+		adminAPI := rpc.NewAdminAPI(bs.driver, bs.Metrics, bs.Log)
+		server.AddAPI(rpc.GetAdminAPI(adminAPI))
+		bs.Log.Info("Admin RPC enabled")
+	}
+	bs.Log.Info("Starting JSON-RPC server")
+	if err := server.Start(); err != nil {
+		return fmt.Errorf("unable to start RPC server: %w", err)
+	}
+	bs.rpcServer = server
+	return nil
+}
+
+// Check ensures that the [BatcherService] is valid
+func (bs *BatcherService) Check() error {
+
+	return nil
+}
```

Comment:
Is it planned to add something here soon? Otherwise I'd delete.

Reply:
Ah, it used to have the rollup/channel config checks, but I moved those to check at initialization time, so the configs are checked before they are used. Will drop the emtpy function